### PR TITLE
Technical SEO - Add /sitemap.xml with hreflang annotations and Sitemap directive in r…

### PR DESCRIPTION
**Pull request by slightlyoffbeat**
## Summary

Improves Mozilla's SEO signal to Google by adding:

1. **`/sitemap.xml`** as the canonical sitemap URL (alongside the existing `/all-urls.xml` which stays working as an alias).
2. **Hreflang annotations** (`<xhtml:link rel="alternate">`) per URL entry, filtered to the supported-locale set sourced from Wagtail's `Locale` model.
3. **`Sitemap:` directive** in robots.txt pointing to the new canonical URL.

Old `/all-urls.xml` had zero hreflang annotations — Google had no machine-readable signal at the sitemap level about which URLs are translations of each other. This change provides that signal explicitly.

Part of the firefox.com technical SEO workstream (item A4 in the master plan). A14b (explicit AI crawler welcome in robots.txt) was scoped in but dropped from this PR — the optics of an explicit named-bot welcome don't justify the zero functional change (AI bots already crawl under `user-agent: *`).

## Why hreflang in the sitemap matters

Today's `canonical-url.html` emits hreflang annotations in the HTML `<head>` of each page. The sitemap, however, was a flat list of URLs with no language relationships declared. Google reads both signals; adding hreflang to the sitemap reinforces the in-page signal and is the convention Google recommends for multi-locale sites.

## The supported-locale filter

Firefox.com has started to use smartling for translations. (~15 supported locales). There are ~88 locales that still have indexed content but it has drifted from the supported content — declaring them as hreflang alternates of supported URLs would misuse hreflang (which is meant for equivalent translations, not drifted ones).

The filter uses **Wagtail's `Locale` model** (`Locale.objects.values_list('language_code', flat=True)`) as the source of truth — the same set Mozilla manages via the Wagtail admin's Settings > Languages page. When Mozilla adds/removes a supported locale via the admin, the sitemap filter updates automatically with no code change.

**Verify before merge:** `Locale.objects.count()` returns the expected ~15 supported locales. If it includes surprise extras, the filter signal would be wrong.

**Legacy-locale behavior:** legacy URLs still appear in their per-locale sitemap for crawl coverage, but emit no `<xhtml:link>` annotations. They become standalone in Google's view — exactly what Mozilla wants given the content drift.

## What changed

**`springfield/sitemaps/urls.py`** — new `sitemap(-global)?.xml` route alongside existing `all-urls(-global)?.xml` (kept as backward-compat alias since Google has indexed it).

**`springfield/sitemaps/views.py`** — `SitemapView` now builds an `alternates` context dict via two new helpers: `_get_supported_locales()` (queries Wagtail Locale.objects) and `_build_alternates()` (groups SitemapURL rows by path, filters to supported locales). NO_LOCALE sitemap and legacy-locale sitemaps both get an empty alternates dict → template emits no `<xhtml:link>` for them.

**`springfield/sitemaps/templates/sitemap.xml`** — adds `<xhtml:link rel="alternate" hreflang="..." href="..."/>` per URL when 2+ supported locales exist for that path. Bare-language conventions (pt-BR → `pt` + `pt-BR`, en-US → `en` + `en-US`, etc.) mirror `canonical-url.html` exactly. pt-PT correctly emits only `pt-PT` (not bare `pt`).

**`springfield/sitemaps/templates/sitemap_index.xml`** — index now emits `/sitemap-global.xml` and `/{locale}/sitemap.xml` URLs.

**`springfield/base/templates/base/robots.txt`** — adds `Sitemap: https://www.firefox.com/sitemap.xml` directive, wrapped in the existing `{% if not disallow_all %}` so non-prod hosts skip it.

**`springfield/settings/base.py`** — adds `/sitemap.xml` and `/sitemap-global.xml` to `SUPPORTED_LOCALE_IGNORE` so the locale middleware doesn't redirect them.

## Tests

`springfield/sitemaps/tests/test_views.py`:
- Updated 4 existing tests (`test_index`, `test_none`, `test_locales`, `test_post`) to cover both legacy and new URLs
- Added 7 new tests covering supported-locale hreflang emission, bare-language conventions for en-US and pt-BR, pt-PT's no-bare-pt convention, single-locale-URL edge case, legacy-locale skip, and the filter excluding legacy from supported-page alternates

`springfield/base/tests/test_views.py`:
- Added `test_robots_production_emits_sitemap_directive` and `test_robots_non_production_omits_sitemap`

**24/24 sitemap tests pass. 7/7 robots tests pass. No regressions in broader `springfield/base/tests/` or `lib/l10n_utils/tests/`.**

## Manual verification after deploy to stage

- `curl https://stage.../sitemap.xml` → 200, sitemap index XML
- `curl https://stage.../en-US/sitemap.xml` → 200, contains `<xhtml:link>` alternates with bare-language conventions (e.g., both `hreflang="en"` and `hreflang="en-US"`)
- `curl https://stage.../robots.txt` → contains `Sitemap: https://www.firefox.com/sitemap.xml`
- `curl https://stage.../{legacy-locale}/sitemap.xml` → 200, URLs listed but no `<xhtml:link>` annotations
- `curl https://stage.../all-urls.xml` → still 200 (backward-compat alias)
- Paste rendered sitemap into Google Search Console Sitemaps validator → no errors

## Follow-ups to file

1. **Extend supported-locale filter to canonical-url.html** (in-page hreflang). Same `Locale.objects`-based approach. Currently in-page hreflang declares all locales where content exists, including drifted legacy content — same misuse as sitemap had before this PR. Mechanism is the same; just needs the template change.
2. **Resubmit `/sitemap.xml` in Google Search Console** post-deploy so Google adopts the new canonical entry point faster (vs. relying on continued use of `/all-urls.xml`).
3. **Monitor Search Console's hreflang validation** for 2–4 weeks post-deploy. Expect transient warnings as Google reindexes; should clear naturally.

## Known transition costs

- Search Console may show transient hreflang warnings during the 2–4 week reindex window (old/new signals coexist). Will clear.
- Slight short-term ranking volatility on legacy-locale URLs as they lose alternate-cluster signal — aligns with strategic direction (legacy slated for sunset).
- Sitemap-vs-in-page inconsistency until follow-up #1 ships: sitemap declares ~15 alternates, in-page hreflang declares ~100. Google reads both. File the follow-up explicitly to avoid drift.

### CODE REVIEW

File: springfield/sitemaps/templates/sitemap.xml [hunk 1]
```diff
@@ -3,6 +3,33 @@
...
+    {%- elif alt_locale == 'hi-IN' %}
+    <xhtml:link rel="alternate" hreflang="hi" href="{{ settings.CANONICAL_URL }}/hi-IN{{ path.path }}"/>
+    {%- elif alt_locale == 'hy-AM' %}
+    <xhtml:link rel="alternate" hreflang="hy" href="{{ settings.CANONICAL_URL }}/hy-AM{{ path.path }}"/>
+    {%- elif alt_locale == 'pa-IN' %}
+    <xhtml:link rel="alternate" hreflang="pa" href="{{ settings.CANONICAL_URL }}/pa-IN{{ path.path }}"/>
+    {%- elif alt_locale == 'sv-SE' %}
+    <xhtml:link rel="alternate" hreflang="sv" href="{{ settings.CANONICAL_URL }}/sv-SE{{ path.path }}"/>
+    {%- elif alt_locale == 'zh-CN' %}
+    <xhtml:link rel="alternate" hreflang="zh" href="{{ settings.CANONICAL_URL }}/zh-CN{{ path.path }}"/>
```
[the remark this answers is not in the document]
Comment (slightlyoffbeat) [reply springfield/sitemaps/templates/sitemap.xml:29]: This duplication is deliberate and called out as a known trade-off in the PR description:

> *Hreflang bare-language logic duplicated between `canonical-url.html` and `sitemap.xml`. Mitigation: add a comment in both templates cross-referencing each other; consider a follow-up refactor to a shared Jinja macro or Python helper. **Not blocking for this PR.***

Refactoring to a shared helper makes sense but is materially bigger scope:
- Both templates would need to be touched
- The in-page hreflang in `canonical-url.html` also needs to adopt the supported-locale filter from this PR (already listed as Follow-up in the description)
- Best to do the helper extraction at the same time as that filter extension so both templates change once, together

Filing the follow-up explicitly so it doesn't drift. Closing this thread as accepted-with-deferred-fix unless there's a concern with the deferral.

File: springfield/sitemaps/views.py [hunk 2]
```diff
@@ -36,11 +40,43 @@ def get_template_names(self):
...
+        to supported locales only.
+        """
+        if locale == NO_LOCALE or locale not in supported_locales:
+            return {}
+
+        alternates = defaultdict(list)
+        for row in SitemapURL.objects.exclude(locale=NO_LOCALE).values("path", "locale"):
+            if row["locale"] in supported_locales:
+                alternates[row["path"]].append(row["locale"])
+        return dict(alternates)
```
[the remark this answers is not in the document]
Comment (slightlyoffbeat) [reply springfield/sitemaps/views.py:71]: Applied — pushed the locale filter into the query and added `order_by` for deterministic output:

  ```python
  rows = SitemapURL.objects.filter(locale__in=supported_locales).order_by("path", "locale").values("path", "locale")
  for row in rows:
      alternates[row["path"]].append(row["locale"])
  ```

The `path__in` further-narrowing wasn't worth the extra query/list-building for current scale, but the locale filter + ordering is a clear win, fewer rows from DB, stable output. Tests still pass.

Skipping the additional `path__in` narrowing — that would require an extra query to materialize the path list first, and at current scale (~10–50K rows, 1h cache) the locale filter alone is sufficient.

**Comment by stephaniehobson** [comment]
I'm seeing a pretty different list of hreflang and pages in the xml : 

- Pontoon backed pages have hrefs to ~63 languages. CMS backed ones have just CMS langs ~19. (_edit: I see this is expected behavour and is documented for followup_)
- `/sitemap.xml` links to sitemaps in 63 other languages. `/en-US/sitemap.xml` only links to ~19 others.
- `/en-US/sitemap.xml` includes aliases for several languages (e.g. es -> es-ES), those do not get corresponding entries on the page hreflang meta tags.
-  some of the non-CMS lang sitemaps 404 ([example](https://www.firefox.com/km/sitemap.xml)).

I don't know how important it is to Google that we be consistent here. I'm inclined to launch and trust it to sort it out but I'm not an SEO expert.

_Edit: We could add removing unsupported languages (ones with 404ing sitemaps) as a follow up, that would make me feel better about merging this._

**Comment by stephaniehobson** [comment]
I was not able to verify the robots.txt behaviour locally. I was hoping I could with prod-mode but it's not a red-flag that I can't.

**Review by stephaniehobson — APPROVED** [review]

**Comment by slightlyoffbeat** [comment]
@stephaniehobson I appreciate the review. Notes on my reasoning:

The sitemap is intentionally asymmetric: the **index** is a broad discovery signal (lists every locale with content), while **hreflang alternates per URL** is a narrow equivalence signal (only the supported set in `Locale.objects`). This matters for the near-future state where legacy locales will keep ~2–3 Pontoon-translated pages outside the CMS — those need sitemap visibility but shouldn't be declared as equivalents of the supported CMS content.

On your specific bullets:

1. ✅ Expected, per your edit. Same filter for both; CMS just has fewer rows to start with.
2. Intentional asymmetry (above). Filtering the index would lose discoverability of the legacy minimal pages.
3. Same root cause as the Copilot maintainability comment — `sitemap.xml` and `canonical-url.html` read from different data sources. I'll bundle the fix into the follow-up PR that extends the supported-locale filter to in-page hreflang, so both templates change once, consistently.
4. Can't repro `/km/sitemap.xml` 404 locally — returns 200 for me. We can check on stage before deploy.
5. Robots.txt: You can verify locally by spoofing the Host header:
   ```bash
   curl -H "Host: www.firefox.com" http://localhost:8000/robots.txt
   ```
   Should return:
   ```
   user-agent: *
   disallow: /*/etc/

   Sitemap: https://www.firefox.com/sitemap.xml
   ```

TL;DR — I think we can ship as-is. Following up on the canonical-url.html consistency in a separate PR.
